src/cache.rs

```diff
@@ -89,53 +86,57 @@ pub enum CacheFreshness {
 }
 
 impl Cache {
-    pub fn new<S>(url: S, platform: PlatformType) -> Self
+    pub fn new<P>(platform: PlatformType, cache_dir: P) -> Self
     where
-        S: Into<String>,
+        P: Into<PathBuf>,
     {
         Self {
-            url: url.into(),
             platform,
+            cache_dir: cache_dir.into(),
         }
     }
 
-    /// Return the path to the cache directory.
-    pub fn get_cache_dir() -> Result<(PathBuf, PathSource)> {
-        // Allow overriding the cache directory by setting the env variable.
-        if let Ok(value) = env::var(CACHE_DIR_ENV_VAR) {
-            let path = PathBuf::from(value);
-            let (path_exists, path_is_dir) = path
-                .metadata()
-                .map_or((false, false), |md| (true, md.is_dir()));
-            ensure!(
-                !path_exists || path_is_dir,
-                "Path specified by ${} is not a directory",
-                CACHE_DIR_ENV_VAR
+    pub fn cache_dir(&self) -> &Path {
+        &self.cache_dir
+    }
```

Comment:
Was this here before? Since we are always using `self.cache_dir` as the implicit `&Path` anyways, let's replace all usages of `cache_dir` with `cache_dir()`

Reply:
You can see the result of the force push here 🙂 https://github.com/dbrgn/tealdeer/compare/8c8f612e38e247843c4bb243520645e4147ed4eb..2894f4823c3a55a02d5b34f4d50d6077ab1038af (On GitHub, click the "force-pushed" text in the message "dbrgn force-pushed the cache-dir-config branch from 8c8f612 to 2894f48 2 days ago" to see the diff.)

The method was here before. But it is only needed to expose the cache directory towards the outside (because the `cache_dir` field is not accessible). I disagree that we should go through a method everywhere just to access a field with the same name, both from an abstraction level point of view and based on the interest of getting this merged without too much additional bikeshedding.

Reply:
Fair enough 🤷